src/dask_awkward/lib/core.py

```diff
@@ -157,7 +157,7 @@ def key(self) -> Hashable:
         return (self._name, 0)
 
     def _check_meta(self, m: Any) -> Any | None:
-        if not isinstance(m, (MaybeNone, UnknownScalar, OneOf)):
+        if not isinstance(m, (TypeTracerArray, MaybeNone, OneOf)):
```

Comment:
I'd suggest replacing `isinstance(x, UnknownScalar)` with `awkward._nplikes.typetracer.is_unknown_scalar`, because that way you'll not pick up ND typetracer arrays here. Note - unlike much of awkward, this needs to be explicitly imported

Reply:
Looks like we get `awkward._nplikes.typetracer.MaybeNone` instances popping up; should `is_unknown_scalar` check for those?

Reply:
I have thought about this; it depends upon what context. I think we might want another function `is_unknown` or something to handle this case. Do you have any example tracebacks? I assume they're not coming from the changes in awkward/main?

Reply:
Or should we make this function
```
if is_unknown_scalar(m) or isinstance(m, (MaybeNone, OneOf)):
    # good to go
else:
    # bad

Reply:
Here's a traceback
```python
>>> import dask_awkward as dak
>>> ds = dak.from_parquet("s3://ddavistemp/hpq", storage_options={"anon": True})
>>> dak.sum(ds.muons.pt, axis=None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ddavis/software/repos/dask-awkward/src/dask_awkward/lib/reducers.py", line 405, in sum
    return total_reduction_to_scalar(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ddavis/software/repos/dask-awkward/src/dask_awkward/lib/core.py", line 1393, in total_reduction_to_scalar
    return new_scalar_object(graph, name_agg, meta=meta)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ddavis/software/repos/dask-awkward/src/dask_awkward/lib/core.py", line 265, in new_scalar_object
    return Scalar(dsk, name, meta, known_value=None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ddavis/software/repos/dask-awkward/src/dask_awkward/lib/core.py", line 107, in __init__
    self._meta: Any = self._check_meta(meta)
                      ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ddavis/software/repos/dask-awkward/src/dask_awkward/lib/core.py", line 162, in _check_meta
    raise TypeError(f"meta must be a typetracer object, not a {type(m)}")
TypeError: meta must be a typetracer object, not a <class 'awkward._nplikes.typetracer.MaybeNone'>
```

Reply:
After looking at our implementation of `total_reduction_to_scalar` I think this may be an issue on our end!

Reply:
My awkward branch is `agoose77/refactor-drop-unknown-scalar`. `MaybeNone` is coming from something like this:

```python
>>> import dask_awkward as dak
>>> import awkward as ak
>>> ds = dak.from_parquet("s3://ddavistemp/hpq", storage_options={"anon": True})
>>> ds.muons.pt._meta
<Array-typetracer [...] type='?? * var * float64'>
>>> ak.max(ds.muons.pt._meta, axis=None)
MaybeNone(TypeTracerArray(dtype('float64'), shape=()))
```

Reply:
Thanks Doug. We've now merged this into main, although I'm going to rework the unknown-length concept to use None in the near future. We do expect `MaybeNone`, and in this case it would make sense for `axis=None`. We'll all need to figure out how we want to handle this in dask-awkward.